src/bytes/src/bytes.cairo

```diff
@@ -135,6 +144,38 @@ impl BytesImpl of BytesTrait {
         Bytes { size, data }
     }
 
+    fn from_byte_array(mut bytes: ByteArray) -> Bytes {
+        let mut res = BytesTrait::new_empty();
+        while !bytes.data.is_empty() {
+            let value: bytes31 = bytes.data.pop_front().unwrap();
+            res.append_bytes31(value);
+        };
+        // Last elem
+        if bytes.pending_word_len != 0 {
+            let mut val: u256 = bytes.pending_word.into();
+            val = U256BitShift::shl(val, 8 * (32 - bytes.pending_word_len.into()));
+            res.concat(@BytesTrait::new(bytes.pending_word_len, array![val.high, val.low]));
```

Comment:
why not just `append_u256` ?

Reply:
If we append_u256, the size of `Bytes` will increase by 32. So the resulting Bytes object will have trailing 0s. Only appending the `pending_word_len` bytes allows us to use `from_byte_array` and `to_byte_array` as inverses of each other without getting weird 0s added to the end/beginning.